ff/serving/metadata/metadata_test.go

```diff
@@ -0,0 +1,1105 @@
+package metadata
```

Comment:
So some of the resources like FeatureResource, keep track of the trainingsets they are in. This is updated via the Notify call. Can you test that when a new training set is created, if I get the FeatureResource that it has the TrainingSet in it?